src/Query/Builder.php

```diff
@@ -147,18 +157,53 @@ public function whereNotIn($column, $values, $boolean = 'and')
         return $this;
     }
 
-    public function orWhereNotIn($column, $values)
+    public function orWhereJsonContains($column, $values)
+    {
+        return $this->whereJsonContains($column, $values, 'or');
+    }
+
+    public function whereJsonDoesntContain($column, $values, $boolean = 'and')
     {
         $this->wheres[] = [
-            'type' => 'NotIn',
+            'type' => 'JsonDoesntContain',
             'column' => $column,
             'values' => $values,
-            'boolean' => 'or',
+            'boolean' => $boolean,
+        ];
+
+        return $this;
+    }
+
+    public function orWhereJsonDoesntContain($column, $values)
+    {
+        return $this->whereJsonDoesntContain($column, $values, 'or');
+    }
+
+    public function whereJsonLength($column, $operator, $value = null, $boolean = 'and')
+    {
+        [$value, $operator] = $this->prepareValueAndOperator(
+            $value, $operator, func_num_args() === 2
+        );
+
+        if ($this->invalidOperator($operator)) {
+            [$value, $operator] = [$operator, '='];
+        }
+
+        $this->wheres[] = [
+            'type' => 'JsonLength',
+            'column' => $column,
+            'value' => $value,
+            'boolean' => $boolean,
         ];
 
         return $this;
     }
 
+    public function orWhereJsonLength($column, $operator, $value)
+    {
+        return $this->whereJsonDoesntContain($column, $operator, $value = null, 'or');
```

Comment:
`return $this->whereJsonLength($column, $operator, $value = null, 'or');`

Reply:
Good spot